cmd/soroban-cli/src/commands/config/alias.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    collections::HashMap,
+    fs::{self, create_dir_all, OpenOptions},
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+use serde::{Deserialize, Serialize};
+use stellar_strkey::DecodeError;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct Data {
+    ids: HashMap<String, String>,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error("cannot parse contract ID {0}: {1}")]
+    CannotParseContractId(String, DecodeError),
+}
+
+impl Data {
+    pub fn load(config_dir: &Path, alias: Option<&str>) -> Result<Option<Self>, Error> {
```

Comment:
Does alias need to an option here?

Reply:
it doesn't. This was due to porting over the code from the other file.